### DISCUSSION TITLE
Are there any complex projects for reference?

### DISCUSSION THREAD

**Discussion by kirkportas** [TimefoldAI/timefold-solver#78] [General]
I'm evaluating Timefold/Optaplanner for a use case in the scheduling / shift-planning / bin-packing domains, and am looking for more complex references and examples than are available in the quick starts. Are there any publicly available repos with examples of the below? Or other advanced examples I haven't thought of that would be useful for learning and mastering this tool?

- Diagrams for advanced cases
- Multi-stage planning, advanced scheduling, or performance benchmarking

**Comment by ge0ffrey** [comment]
As for examples, these resources should help:

- 9 use-case quickstarts in quickstarts repo: https://github.com/TimefoldAI/timefold-quickstarts/tree/stable/use-cases
- 16 examples in solver repo: https://github.com/TimefoldAI/timefold-solver/tree/main/examples
- Documentation for those 16 examples: https://timefold.ai/docs/timefold-solver/latest/use-cases-and-examples/use-cases-and-examples.html (which includes class diagrams)

On your other questions:
- Muti-stage planning and advanced scheduling: You might find my old OptaPlanner video useful on this topic, we will reshoot it with updated content soon: https://youtu.be/Ew6pq9nJKog
- Performance benchmarking: We have an entire benchmarker tool, see documentation here: https://timefold.ai/docs/timefold-solver/latest/benchmarking-and-tweaking/benchmarking-and-tweaking.html

For the next version, the benchmarker tool got an major improvement.

Hope that helps :)
